### PR TITLE
Basic counter examples

### PR DESCRIPTION
**Pull request by chiubaca**
This is PR provides a simple counter demo implemented with Options API, Class API and Composition API for each of the "Basic" examples at https://typescript.nuxtjs.org/examples/options-api/basic.html

### CODE REVIEW

**Comment by chiubaca** [comment]
Hmm my demo app is not coming through in the code sandbox embeds in the netlify preview page... why?

https://deploy-preview-453--nuxt-typescript.netlify.app/examples/options-api/basic.html

**Comment by kevinmarrec** [comment]
Hi @chiubaca , thanks for your PR, would you mind switching from `npm` to `yarn` for the examples ? (`yarn.lock` instead of `package-lock.json`)

**Comment by chiubaca** [comment]
@kevinmarrec no probs, was for my own learning 😄.  

I've purged the `package-lock.json` files. I generated the `yarn.lock` files on my local machine but noticed there's a `.gitignore` rule at line 5 preventing them from being committed though...

**Comment by kevinmarrec** [comment]
@chiubaca Oh yeah sorry ! That's intended :)

Gonna perform a new review now

File: examples/options-api/basic/pages/index.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+})
+</script>
+
+<style>
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height:100vh;
```
Comment (kevinmarrec) [inline examples/options-api/basic/pages/index.vue:39]:
```suggestion
  height: 100vh;
```

File: examples/composition-api/basic/pages/index.vue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+})
+</script>
+
+<style>
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height:100vh;
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:43]:
```suggestion
  height: 100vh;
```

File: examples/composition-api/basic/pages/index.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    function increment () {
+      counter.value++
+    }
+
+    function decrement () {
+      counter.value--
+    }
+
+    return { counter, increment, decrement }
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:32]:
```suggestion
    return {
      counter,
      increment,
      decrement
    }
```

File: examples/composition-api/basic/pages/index.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  setup () {
+    const counter = ref(0)
+
+    function increment () {
+      counter.value++
+    }
+
+    function decrement () {
+      counter.value--
+    }
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:30]:
```suggestion
    const increment = () => counter.value++
    const decrement = () => counter.value--
```

File: examples/composition-api/basic/pages/index.vue [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import VueCompositionApi, { defineComponent, ref } from '@vue/composition-api'
+
+Vue.use(VueCompositionApi)
+
+export default defineComponent({
+  name: 'CompositionAPICounter',
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:20]: Is this required ?

File: examples/composition-api/basic/pages/index.vue [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      Decrement
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import VueCompositionApi, { defineComponent, ref } from '@vue/composition-api'
+
+Vue.use(VueCompositionApi)
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:17]: Installation of `VueCompositionApi` is done by the Nuxt plugin (`plugins/composition-api.js`)
```suggestion
import { defineComponent, ref } from '@vue/composition-api'
```

File: examples/class-api/basic/pages/index.vue [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}
+</script>
+
+<style>
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height:100vh;
```
Comment (kevinmarrec) [inline examples/class-api/basic/pages/index.vue:36]:
```suggestion
  height: 100vh;
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

File: examples/options-api/basic/pages/index.vue [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      counter: 0
+    }
+  },
+  methods: {
+    increment () {
+      return this.counter++
+    },
+    decrement () {
+      return this.counter--
+    }
```
Comment (kevinmarrec) [inline examples/options-api/basic/pages/index.vue:28]: These functions shouldn't be returning any value.
I suggest also always provide return type of functions in TypeScript projects (even if `void`)
```suggestion
    increment (): void {
      this.counter++
    },
    decrement (): void {
      this.counter--
    }
```

File: examples/composition-api/basic/pages/index.vue [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { defineComponent, ref } from '@vue/composition-api'
+
+export default defineComponent({
+  name: 'CompositionAPICounter',
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:18]: Remove cause we're defining `index.vue` which is page component.
Also Nuxt automatically name component based on file name so it isn't required here
```suggestion
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

File: examples/composition-api/basic/pages/index.vue (hunk 5 above)
Comment (chiubaca) [reply examples/composition-api/basic/pages/index.vue]: ln 17? `Vue.use(VueCompositionApi)` thats been removed in one of the other commits now.

**Comment by chiubaca** [comment]
@kevinmarrec  Noticed the nuxt docs have had revamp since this PR. There used to be these additional sections in the examples:

image

Hope you can still carve a home for these examples somewhere 🙏. I was also interested in helping produce some "Advance" examples too. Just let me know if I can help in any way.

File: examples/composition-api/basic/pages/index.vue [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    </button>
+    <p>Counter : {{ counter }}</p>
+    <button @click="decrement">
+      Decrement
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
```
Comment (kevinmarrec) [inline examples/composition-api/basic/pages/index.vue:14]: Remove this unused import to fix tests
```suggestion
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

**Comment by kevinmarrec** [comment]
@chiubaca Yes, i'm waiting tests to be resolved with my last commit (unused import) then I'm gonna merge.

Finally I'll take time to see how these basic examples can show up again on new documentation :)

**Review by kevinmarrec — APPROVED** [review]

**Comment by kevinmarrec** [comment]
Thanks for your work and time @chiubaca , I'll let you know when it's live on website :)

**Comment by chiubaca** [comment]
yay, thanks for taking the time to review. Learnt a few things :)

**Comment by kevinmarrec** [comment]
@chiubaca It's live on https://typescript.nuxtjs.org/

Still it's only on English language, see my commit :
https://github.com/nuxt/typescript/commit/b8a349fd410d80e6ca25b51b112439595d3e6b10

It needs further work and probably some new translations to have them also on all other languages.

**Comment by chiubaca** [comment]
woohoo! 

With regards to advanced examples, happy to build a simple todo app? or something similar along those lines. Let me know :) happy to sling over a new PR.

**Pull request merged**